pgcli/main.py

```diff
@@ -1011,16 +1012,26 @@ def format_arrays(data, headers, **_):
         headers = [case_function(utf8tounicode(x)) for x in headers]
         rows = list(cur)
         formatted = formatter.format_output(rows, headers, **output_kwargs)
-        first_line = formatted[:formatted.find('\n')]
+
+        if isinstance(formatted, (text_type)):
```

Comment:
This is exactly what this PR is about (you have a sharp eye!). In the next version of cli_helpers a generator will be outputted from format_output which yields lines without newline characters.

When this new version of cli_helpers is released these lines can be removed. See https://github.com/dbcli/cli_helpers/pull/13